tests/unit_tests/udfs/test_base_udf.py

```diff
@@ -0,0 +1,141 @@
+    mock_exa = create_mock_exa_environment(
+        [bucketfs_conn_name],
+        [bucketfs_conn],
+        mock_meta,
+        '',
+        None)
+    mock_ctx = create_mock_udf_context(input_data, mock_meta)
+
+    udf = DummyImplementationUDF(exa=mock_exa,
+                             base_model=mock_base_model_factory,
+                             tokenizer=mock_tokenizer_factory)
+    udf.run(mock_ctx)
+    res = mock_ctx.output
+    # check if no errors
+    assert res[0][-1] is None and len(res[0]) == len(mock_meta.output_columns)
+
+
+@pytest.mark.parametrize("description, bucketfs_conn_name, bucketfs_conn ,"
+                         "sub_dir, model_name", [
+    ("all null", '', None, None, None),
+    ("model name missing", "test_bucketfs_con_name", Connection(address=f"file:///test"),
+     "test_subdir", None),
+    ("bucketfs_conn missing", None, None,
+     "test_subdir", "test_model"),
+    ("sub_dir missing", "test_bucketfs_con_name", Connection(address=f"file:///test"),
+     None, "test_model"),
+    ("model_name missing", "test_bucketfs_con_name", Connection(address=f"file:///test"),
+     "test_subdir", None)
+])
+def test_model_downloader_missing_parameters(description, bucketfs_conn_name, bucketfs_conn, sub_dir, model_name):
+    mock_base_model_factory: Union[ModelFactoryProtocol, MagicMock] = create_autospec(ModelFactoryProtocol)
+    mock_tokenizer_factory: Union[ModelFactoryProtocol, MagicMock] = create_autospec(ModelFactoryProtocol)
+
+    mock_bucketfs_factory: Union[BucketFSFactory, MagicMock] = create_autospec(BucketFSFactory)
+    mock_bucketfs_locations = [Mock()]
+    mock_cast(mock_bucketfs_factory.create_bucketfs_location).side_effect = mock_bucketfs_locations
+
+    input_data = [
+        (
+            1,
+            model_name,
+            sub_dir,
+            bucketfs_conn_name,
+            ''
+        ),
+        (
+            1,
+            model_name,
+            sub_dir,
+            bucketfs_conn_name,
+            ''
+        )
+    ]
+    mock_meta = create_mock_metadata()
+    mock_exa = create_mock_exa_environment(
+        [bucketfs_conn_name],
+        [bucketfs_conn],
+        mock_meta,
+        '',
+        None)
+    mock_ctx = create_mock_udf_context(input_data, mock_meta)
+
+    udf = DummyImplementationUDF(exa=mock_exa,
+                                 base_model=mock_base_model_factory.side_effect,
+                                 tokenizer=mock_tokenizer_factory.side_effect)
+
+    udf.run(mock_ctx)
+    res = mock_ctx.output
+    print(res)
+    error_field = res[0][-1]
+    expected_error_start = f"For each model model_name, bucketfs_conn and sub_dir need to be provided. " \
+                           f"Found model_name = {model_name},"
+    expected_error_end = f" sub_dir = {sub_dir}"
+    assert error_field is not None and len(res[0]) == len(mock_meta.output_columns)
+    assert expected_error_start in error_field and expected_error_end in error_field
```

Comment:
you could use regex_matcher , not sure if we have it already in this project, otherwise take a look in other projects